src/plotly/chartDefinition.ts

```diff
@@ -128,28 +131,72 @@ const getChartData = (
         ),
       };
     }
-
-    const newSeries = {
-      ...trace,
-      stackgroup: chartType === "stacked filled area" ? "one" : undefined,
-      name: series.name,
-      type: chartType === "stacked bar" ? "bar" : chartType,
-      mode: chartProps?.LegendSection?.mode ?? "lines",
-      hoverinfo: chartProps.Interactivity.interactivity,
-      marker: { color: series.color },
-      line: {
-        color: series.color,
-        dash: series.dashStyle,
-      },
-      fill:
-        chartType === "filled area" || chartType === "stacked filled area"
-          ? "tonexty"
-          : "none",
-    };
-    isAStackedBar ? traces.unshift(newSeries) : traces.push(newSeries);
+    let newSeries = {};
+    if (series.intervalType === "intervals") {
```

Comment:
could use a comment on this conditional, this is where the magic happens (I think) and future you/us/new-friends will thank you 😄

Reply:
Made a comment explaining in more detail